src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -284,7 +284,10 @@ internal IOperation CreateInternal(BoundNode boundNode)
                     return CreateRangeExpressionOperation((BoundRangeExpression)boundNode);
                 case BoundKind.SwitchSection:
                     return CreateBoundSwitchSectionOperation((BoundSwitchSection)boundNode);
+                case BoundKind.UnconvertedConditionalOperator:
+                    throw ExceptionUtilities.Unreachable;
                 case BoundKind.UnconvertedSwitchExpression:
+                    throw ExceptionUtilities.Unreachable;
```

Comment:
Not sure how this would be unreachable. What if someone speculates on a switch expression by itself, and then attempts to get the IOperation tree for it? Please add some tests. Same for unconverted conditional operators.

Reply:
There is no API that I can find to speculate on just an expression. Please be more specific about the scenario you'd like tested.

---
In reply to: [410435964](https://github.com/dotnet/roslyn/pull/40604#discussion_r410435964) [](ancestors = 410435964)

Reply:
Nothing would stop someone from creating an `ExpressionStatement` with just a ternary in it and speculating on that statement.

---
In reply to: [410480904](https://github.com/dotnet/roslyn/pull/40604#discussion_r410480904) [](ancestors = 410480904,410435964)